File: Reqnroll/Chapter13/13.4-EntitySelectorPattern/Before/WIMP.Specs/Support/CustomParameterTypes.cs
using Reqnroll;

using WIMP.App.Services;

namespace WIMP.Specs.Support;

[Binding]
public class CustomParameterTypes
{
    [StepArgumentTransformation(@"the order #(\d+)", Name = "order")]
    public int ConvertOrderNumber(int orderNo)
    {
        var order = OrderService.GetOrder(orderNo);
        return order?.OrderNo ??
               throw new InvalidOperationException("Order not found");
    }
}

File: CucumberJVM/Chapter14/14.5-DataTableAccessorPattern/Before/src/test/java/com/wimp/app/specs/support/CustomParameterTypes.java
package com.wimp.app.specs.support;

import io.cucumber.java.ParameterType;

import java.math.BigDecimal;

public class CustomParameterTypes {

    @ParameterType("\\$?(\\d+(?:\\.\\d{1,2})?)")
    public BigDecimal price(String value){
        return new BigDecimal(value);
    }
}
